entry_types/scrolled/package/spec/frontend/AudioPlayer/structuredData-spec.js

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+
+import {renderInEntry} from 'support';
+import {getInitialPlayerState, getPlayerActions} from 'support/fakePlayerState';
+import {useFakeMedia} from 'support/fakeMedia';
+import * as jsonLdQueries from 'support/jsonLdQueries';
+
+import {AudioPlayer} from 'frontend/AudioPlayer';
+
+describe('AudioPlayer structured data', () => {
+  useFakeMedia();
+
+  function requiredProps() {
+    return {
+      playerState: getInitialPlayerState(),
+      playerActions: getPlayerActions()
+    };
+  }
+
+  it('is rendered by default', () => {
+    const {getJsonLd} =
+      renderInEntry(<AudioPlayer {...requiredProps()} id={100} />, {
+        seed: {
+          fileUrlTemplates: {
+            audioFiles: {
+              mp3: '//cdn/:id_partition/:basename.mp3'
+            }
+          },
+          entry: {
+            publishedAt: '2020-08-10',
+          },
+          audioFiles: [
+            {
+              id: 1,
+              permaId: 100,
+              createdAt: '2020-08-01',
+              isReady: true,
+              basename: 'audio',
+              durationInMs: (3 * 60 + 43) * 1000 + 120,
+              rights: 'some author',
+              configuration: {alt: 'alt text'}
+            }
+          ]
+        },
+        queries: jsonLdQueries
+      });
+
+    expect(getJsonLd()).toMatchObject({
+      '@context': 'http://schema.org',
+      '@type': 'AudioObject',
+      name: 'audio',
+      description: 'alt text',
+      url: 'https://cdn/000/000/001/audio.mp3',
+      duration: 'PT3M43S',
+      datePublished: '2020-08-10',
```

Comment:
**thought:** Adding this here, because we didn't touch article markup in this commit: Currently, all Pageflows receive a `datePublished` in schema markup. This means that Google is going to mention this date in search snippets [1]. For Pageflows that are more timeless, we could consider making `datePublished` and/or `dateModified` optional. Currently, in the `structured_data/entry` template, we always set `datePublished`, as well as `dateModified` [2]

[1] https://developers.google.com/search/docs/data-types/article#non-amp-sd
[2] https://github.com/codevise/pageflow/blob/1f1a694151db27a0d5a535c279a0ffea78de72bb/app/views/pageflow/structured_data/_entry.json.jbuilder#L30

Reply:
Yeah, there are a lot of opportunities to improve the actual structured data being rendered here. The original requirements regarding included attributes came from the customer. This PR merely tries to bring Scrolled to the same level of support that is currently present for Paged.

This also answers the question regarding before/after. There is also no structured data for video or chart embeds at the moment.